simphony_mayavi/sources/tests/test_slim_cuds_source.py

```diff
@@ -0,0 +1,555 @@
+                                point_scalars="MASS", point_vectors="")
+
+        self.assertEqual(source.point_scalars_name, "MASS")
+        self.assertEqual(source.point_vectors_name, "")
+
+        source.start()
+
+        self.assertEqual(source.point_scalars_name, "MASS")
+        self.assertEqual(source.point_vectors_name, "")
+
+
+class TestLatticeSlimSource(TestLatticeSource):
+    tested_class = SlimCUDSSource
+
+    def test_source_from_a_cubic_lattice(self):
+        cuds = make_cubic_lattice('test', 0.4, (14, 24, 34), (4, 5, 6))
+        self._add_velocity(cuds)
+        source = SlimCUDSSource(cuds=cuds)
+        self.assertEqual(source.data.number_of_points, 14 * 24 * 34)
+        assert_array_equal(source.data.origin, (4.0, 5.0, 6.0))
+
+        self.assertEqual(source.data.point_data.number_of_arrays, 1)
+        self.assertEqual(len(source._point_vectors_list), 2)
+        self.assertEqual(source._point_vectors_list, ['VELOCITY', ''])
+        self.assertEqual(source.point_vectors_name, 'VELOCITY')
+
+        source.point_vectors_name = ""
+        self.assertEqual(source.data.point_data.number_of_arrays, 0)
+
+        source.point_vectors_name = 'VELOCITY'
+        self.assertEqual(source.data.point_data.number_of_arrays, 1)
+        vectors = source.data.point_data.vectors.to_array()
+
+        for node in cuds.iter_nodes():
+            point_id = source.data.compute_point_id(node.index)
+            assert_array_equal(
+                cuds.get_coordinate(node.index),
+                source.data.get_point(point_id))
+            assert_array_equal(vectors[point_id], node.index)
+
+        source.point_vectors_name = ''
+        self.assertEqual(source.data.point_data.number_of_arrays, 0)
+
+    def test_source_from_an_orthorombic_p_lattice(self):
+        cuds = make_orthorhombic_lattice(
+            'test',  (0.5, 0.54, 0.58), (15, 25, 35), (7, 9, 8))
+        self._add_velocity(cuds)
+        source = SlimCUDSSource(cuds=cuds)
+        self.assertEqual(source.data.number_of_points, 15 * 25 * 35)
+        assert_array_equal(source.data.origin, (7.0, 9.0, 8.0))
+
+        self.assertEqual(source.data.point_data.number_of_arrays, 1)
+        self.assertEqual(len(source._point_vectors_list), 2)
+        self.assertEqual(source._point_vectors_list, ['VELOCITY', ''])
+        self.assertEqual(source.point_vectors_name, 'VELOCITY')
+
+        source.point_vectors_name = ''
+        self.assertEqual(source.data.point_data.number_of_arrays, 0)
+
+        source.point_vectors_name = 'VELOCITY'
+        self.assertEqual(source.data.point_data.number_of_arrays, 1)
+
+        vectors = source.data.point_data.vectors.to_array()
+        for node in cuds.iter_nodes():
+            point_id = source.data.compute_point_id(node.index)
+            assert_array_equal(
+                cuds.get_coordinate(node.index),
+                source.data.get_point(point_id))
+            assert_array_equal(vectors[point_id], node.index)
+
```

Comment:
lines 453-508 look to me codes that can be refactored for SlimCUDSSource and CUDSSource test cases.  But I think we could make a separate refactoring issue/PR for this.